Swift Sources/UIViewController+Promise.swift

```diff
@@ -19,6 +20,15 @@ class MFMailComposeViewControllerProxy: NSObject, MFMailComposeViewControllerDel
         }
         PMKRelease(self)
     }
+
+    func messageComposeViewController(controller: MFMessageComposeViewController!, didFinishWithResult result: MessageComposeResult) {
+        if result.value == MessageComposeResultFailed.value {
+            controller.reject(NSError(domain: PMKErrorDomain, code: 4, userInfo: [NSLocalizedDescriptionKey: "The user’s attempt to save or send the message was unsuccessful."]))
```

Comment:
Ok yes I wasn't sure what error code to use here so I used 4 because I didn't see it used for anything in the swift implementation.